src/main/java/org/codehaus/mojo/versions/utils/PropertiesVersionsFileReader.java

```diff
@@ -0,0 +1,59 @@
+package org.codehaus.mojo.versions.utils;
+
+import org.codehaus.mojo.versions.Property;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+public class PropertiesVersionsFileReader {
+
+    /**
+     * Commas-separated list of properties
+     */
+    private String propertiesCsv;
+
+    private Property[] propertiesConfig;
+
+    private String propertyFilePath;
+
+    public PropertiesVersionsFileReader(String filePath) {
+        propertyFilePath = filePath;
+    }
+
+    public void read() throws IOException {
+        try (InputStream input = new FileInputStream(propertyFilePath)) {
+
+            Properties prop = new Properties();
+
+            // load a properties file
+            prop.load(input);
+
+            prop.propertyNames();
+
+            propertiesCsv = prop.keySet().stream().map(Object::toString).collect(Collectors.joining(","));
+
+            List<Property> propertiesConfigList = new ArrayList<>();
+            prop.forEach((name, version) ->  {
+                Property propertyConfig = new Property((String)name);
+                propertyConfig.setVersion((String)version);
+                propertiesConfigList.add(propertyConfig);
+            });
+
+            propertiesConfig = propertiesConfigList.toArray(new Property[0]);
```

Comment:
what about something such 
```
            StreamSupport.stream(prop.entrySet().spliterator(), false).map(objectObjectEntry -> {
                Property propertyConfig = new Property((String)objectObjectEntry.getKey());
                propertyConfig.setVersion((String)objectObjectEntry.getValue());
                return propertyConfig;
            }).collect(Collectors.toList()).toArray(new Property[0]);
```

Reply:
@olamy you can directly convert into an array, no need to collect it to a list first.
Also, I would like to see the lambda method refactored out for readability. And I guess this code is also used elsewhere.
Rationale: a 1-line map call is much easier to read than a three line lambda.